### ISSUE TITLE
Output loss that depends on the expected value other outputs

### ISSUE THREAD

**Issue by damip** [keras-team/keras#9679]
Let's imagine we have a face localization model with one input (an image), and two outputs:
- coord_pred : a vector of coordinates giving the location of the face
- present_pred : sigmoid output saying whether or not a face is present in the image

We want the error to be : 
`binary_crossentropy( present_expected, present_pred ) + present_expected * mean_squared_error( coord_expected, coord_pred )`
so that the error on the coordinates is always zero when there is no face in the image.

The current keras implementation allows the definition of custom loss functions for each output, but these functions are called by keras with y_true and y_pred of the corresponding output as the sole parameters. Therefore, the loss functions cannot have access to the y_true of the other outputs for conditional weighting.

As for the `sample_weight` parameter of `Model.fit`, it seems that it does not achieve this purpose, as it is applied to the total loss...

Any ideas ?
